chain/chain/src/flat_storage_creator.rs

```diff
@@ -423,13 +422,16 @@ impl FlatStorageCreator {
         let mut creation_needed = false;
         for shard_id in 0..num_shards {
             if runtime_adapter.cares_about_shard(me, &chain_head.prev_block_hash, shard_id, true) {
-                let status = runtime_adapter.try_create_flat_storage_state_for_shard(
-                    shard_id,
-                    chain_store.head().unwrap().height,
-                    chain_store,
-                );
+                let status = runtime_adapter.get_flat_storage_creation_status(shard_id);
                 match status {
-                    FlatStorageStateStatus::Ready | FlatStorageStateStatus::DontCreate => {}
+                    FlatStorageCreationStatus::Ready => {
+                        runtime_adapter.create_flat_storage_state_for_shard(
+                            shard_id,
+                            chain_store.head().unwrap().height,
```

Comment:
can't we use `chain_head` variable instead of `chain_store.head().unwrap()` here?

also it would be nice to replace that `unwrap` with `.except(<Message explaining why we expect head to be present>)` or add a comment on top explaining why `.unwrap()` wouldn't fail here

Reply:
Yeah. It's even better to make `FlatStorageCreator::new` return `Result<Option<Self>, Error>` and propagate error if it occurs.